src/client.rs

```diff
@@ -101,21 +103,39 @@ impl Client {
     }
 
     /// Run post request towards API
-    pub(crate) async fn post_request_with_body<B>(
+    pub(crate) async fn post_request_with_body<B: Body>(
         &self,
         endpoint: String,
         body: B,
     ) -> APIResult<Response>
     where
         B: Sized + serde::Serialize,
     {
+        let inner_body = body.to_inner(self.options.api_key.clone());
         self.client
-            .get(self.full_url(endpoint))
+            .post(self.full_url(endpoint))
             .header(CONTENT_TYPE, "application/json")
-            .header(AUTHORIZATION, format!("Bearer {}", self.options.api_key))
-            .body(serde_json::to_string(&body).expect("unwrap here is safe"))
+            .body(inner_body.expect("unwrap here is safe"))
```

Comment:
should we improve this and remove the expect?

Reply:
Yes, that'd be neat!